pandas/sparse/array.py

```diff
@@ -322,31 +332,51 @@ def take(self, indices, axis=0):
             # return scalar
             return self[indices]
 
-        indices = np.atleast_1d(np.asarray(indices, dtype=int))
-
-        # allow -1 to indicate missing values
+        indices = com._ensure_platform_int(indices)
```

Comment:
I realize that *maybe* could move ``_assert_take_fillable`` (and the shared docs for it), to another Mixin (so can use in ``Sparse``). maybe.